test/style/pipes_test.exs

```diff
@@ -498,6 +498,13 @@ defmodule Styler.Style.PipesTest do
       assert_style("a |> (fn x -> x end).() |> c", "a |> then(fn x -> x end) |> c()")
     end
 
+    test "rewrites then/2 when the passed function is a named function reference" do
+      assert_style("a |> then(&fun/1) |> c", "a |> fun() |> c()")
+      assert_style("a |> then(&DateTime.from_is8601/1) |> c", "a |> DateTime.from_is8601() |> c()")
+      assert_style("a |> then(&DateTime.from_is8601/1)", "DateTime.from_is8601(a)")
+      assert_style("a |> then(&fun(&1, d)) |> c", "a |> then(&fun(&1, d)) |> c()")
```

Comment:
this could actually also be rewritten as `a |> fun(d) |> c()`

Reply:
saw you reverted the commit that _did_ do that - did you see an edge case i'm not? thinking i'll just revert your revert after reimplementing the exact same function 😂

Reply:
yeah i am afraid it doesn't handle the following case:

```
"a |> then(&Map.put(&1, :attribute, &1.other_attribute)) |> c()"
```

as it produces the following AST:

```
{{:., [line: 1], [{:__aliases__, [last: [line: 1], line: 1], [:Map]}, :put]},
 [
   {:__block__, [line: 1], [:attribute]},
   {{:., [line: 1], [{:&, [line: 1], [1]}, :other_attribute]},
    [no_parens: true, line: 1], []}
 ]}
 ```

i think it would take another clause or some massaging of the `args` in https://github.com/adobe/elixir-styler/pull/126/commits/59febab7504f6d20f60a450d7717bd83d81e6d02 to make it work properly

Reply:
ah thanks! can handle that by verifying that `{&, _, [1]}` doesn't appear anywhere else in the tree

Reply:
yes! I am happy to spin up a second PR for this later today

Reply:
https://github.com/adobe/elixir-styler/commit/b01d32835c614921129da47844c576784d150923 if you're curious =)

Reply:
🙌 great! Thanks @novaugust :)

Reply:
(hah sorry for the poor communication, meant to say "i'll whip that out in a jiffy" but obviously forgot in my enthusiasm to get started)

thanks again for the great idea, you really nailed the spirit of styler ❤️